components/marketplace/buy-fractional-order-form.tsx

```diff
@@ -155,17 +155,15 @@ export const BuyFractionalOrderForm = ({
   const percentageAmount = form.watch("percentageAmount");
   const pricePerPercent = form.watch("pricePerPercent");
 
-  const unitsToBuy =
-    BigInt(getUnitsToBuy(percentageAmount)) > BigInt(0)
-      ? getUnitsToBuy(percentageAmount)
-      : "0";
+  const unitsToBuy = getUnitsToBuy(percentageAmount);
+  const pricePerUnit = getPricePerUnit(
+    pricePerPercent,
+    BigInt(hypercert.units || 0),
+  );
 
   const totalPrice = formatPrice(
     order.chainId,
-    getTotalPriceFromPercentage(
```

Comment:
Looks like this was the only reference to `getTotalPriceFromPercentage()`, so it can be removed and its tests. Can you confirm that?

Reply:
The above is not a merge blocker. Can be dealt with in a follow-up.